### DISCUSSION TITLE
Redirect crash when source contain query param

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#29701] [Help] [answered]
Hi, 
I am trying to setup redirect in next.config.js.
The source is "/abc?d=e"
the destination is "/gh"
But currently I am unable to do so as server keep crashing with error `source` parse failed for route ;
Any idea whats going wrong.

**Comment by ijjk** [comment, marked as answer]
Hi, it sounds like you are looking for `has` matching which allows you to match `query` values for a redirect. Note: by default the query is not matched in the `source` value and should only be the pathname to match against. Related documentation is available [here](https://nextjs.org/docs/api-reference/next.config.js/redirects#header-cookie-and-query-matching)

**Reply by ghost** [reply]
Hi, @ijjk Thanks for providing the answer. Now, the redirection seems to work but now I have one another issue. When I redirect "/abc?d=e" -> "/gh", using the following setting:
` async redirects() {
    return [
      {
        source: "/abc",
        has: [
          {
            type: "query",
            key: "d",
            value: "e",
          },
        ],
        permanent: false,
        destination: "/gh",
      },
    ];`

it redirects to "/gh?d=e". which will cause a bit of an issue with analytics I want it to exactly redirect it to "/gh" without any query string.
Any idea, on what seems wrong in the above config?

**Reply by ghost** [reply]
@ijjk any update?

**Reply by ijjk** [reply]
Why does this cause an issue with analytics? It is expected to pass through query values so that values that are used by analytics like `utm_source` aren't lost during a redirect.

**Reply by ghost** [reply]
@ijjk Well the problem is that it should exactly to the '/gh' page according to SEO team and as 'gh' page is including the query string its causes issue where the traffic is coming to '/gh?d=e' instead of '/gh' page. The query string is of no significance on this 'gh' page it doesn't even use the query string so why its getting included. Isn't there any option or way to remove the query string?

**Reply by ijjk** [reply]
You can rewrite to an API route that handles the redirect instead if you must remove the query although this should not affect SEO.
